test/functional/test_framework/test_framework.py

```diff
@@ -559,24 +557,26 @@ def wait_for_node_exit(self, i, timeout):
         self.nodes[i].process.wait(timeout)
 
     def connect_nodes(self, a, b):
-        id_a = self.nodes[a].p2p_get_peer_id()
-        id_b = self.nodes[b].p2p_get_peer_id()
+        count_a = self.nodes[a].p2p_get_peer_count()
+        count_b = self.nodes[b].p2p_get_peer_count()
```

Comment:
Have we lost the ability to request peer ID via RPC? If so, it looks like a functionality regression. Do you intend to re-add it or provide an alternative way of identifying peers of given node?

Reply:
There are two RPC calls: `p2p_get_peer_id` and `get_connected_peers`.
`p2p_get_peer_id` (used to return globally unique local node id) is gone forever because peer ID is now a simple counter. This RPC call become meaningless (at least in the mock backend).
`get_connected_peers` can be updated later to also return this new id, for example to allow disconnect specific peers. Currently it's possible to disconnect peers by their remote socket address.

Reply:
count is not like id. `count` is supposed to be "the number of peers available right now".

Reply:
This test shouldn't be changed at all. The unique id per connection should be used to identify nodes at the connection level. For some reason we seem to be doing an "all or nothing" thing...

Reply:
> count is not like id. `count` is supposed to be "the number of peers available right now".

Yes, count is the number of connected peers.
We check how many nodes was connected before and wait until we get expected count.
Sorry, I don't understand what is supposed to be wrong here.

> This test shouldn't be changed at all. The unique id per connection should be used to identify nodes at the connection level. For some reason we seem to be doing an "all or nothing" thing...

It's not possible to proceed without updating the test.
The test worked before because it's assumed that the peer ids are globally unique.
It was OK for libp2p where public key from noise was used to identify the nodes, but it was a hack in the mock backed, where peer id was derived from the socket address.
As it was decided we are not going to use globally unique ids and so the tests should not assume that we can somehow guess our node id when we connect to some other peer. `id_a` and `id_b` here was supposed to be such guessed id and I was forced to change it.

Reply:
I understand what you mean and I see where this misunderstanding is coming from. The test is using ids, which is not a good handle. There should not be usage of ids in the test, but the object of the node should be used. Then, to communicate with the node specifically, the id should be used in the context of one connection.

So let's say NodeA, for testing purposes, needs to send a specific message to NodeB. This is how it should work in functional tests:

- You call `node_b_id = NodeB.get_peer_id()`
- You call `NodeA.send_my_special_message(node_b_id)`

This way, you shouldn't care about the address. Dealing with addresses is messy at the test level and is its own abstraction layer. The only practical purpose of addresses, besides the obvious communication, is banning addresses and associating behavior. But we shouldn't be throwing strings all over the place in our interfaces. That's playing with fire.

Reply:
Sorry, I still don't see how it's relevant to my changes. I removed `get_peer_id` from RPC because after recent discussion it  was decided that we are not going to have global ids. Without global ids `get_peer_id` becomes meaningless (what can we return there?). So I removed it and I updated the functional tests that depend on it (`connect_nodes` and `disconnect_nodes` helper functions). That functions also assumed that the addresses are from libp2p which I also fixed.